### PR TITLE
[Prow CI Step Enablement] MGMT-12390: add step to Config mgmt cluster dns

### PR DESCRIPTION
**Pull request by LiangquanLi930**
add step: `hypershift-agent-create-config-dns` to Config mgmt cluster dns

### CODE REVIEW

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Comment by LiangquanLi930** [comment]
https://hypershift-docs.netlify.app/how-to/agent/create-agent-cluster/#metal3

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

File: ci-operator/step-registry/hypershift/agent/create/hostedcluster/hypershift-agent-create-hostedcluster-commands.sh [hunk 1]
```diff
@@ -17,6 +17,7 @@ oc extract secret/pull-secret -n openshift-config --to=/tmp --confirm
   --pull-secret=/tmp/.dockerconfigjson \
   --agent-namespace="clusters-${CLUSTER_NAME}" \
   --base-domain=${BASEDOMAIN} \
+  --api-server-address=api.${CLUSTER_NAME}.${BASEDOMAIN} \
```
Comment (eranco74) [inline ci-operator/step-registry/hypershift/agent/create/hostedcluster/hypershift-agent-create-hostedcluster-commands.sh:20]: I never tried this, so I just get the IP of the host with the node port. 
I see that you added DNS for:
api.\$HOSTEDCLUSTER_NAME.ostest.test.metalkube.org

Is "ostest.test.metalkube.org" the base domain?

File: ci-operator/step-registry/hypershift/agent/create/config-dns/hypershift-agent-create-config-dns-commands.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+CLUSTER_NAME="$(echo -n $PROW_JOB_ID|sha256sum|cut -c-20)"
+echo "$CLUSTER_NAME" > /tmp/hostedcluster_name
+scp "${SSHOPTS[@]}" "/tmp/hostedcluster_name" "root@${IP}:/home/hostedcluster_name"
+
+# shellcheck disable=SC2087
+ssh "${SSHOPTS[@]}" "root@${IP}" bash - << EOF |& sed -e 's/.*auths.*/*** PULL_SECRET ***/g'
+MASTER_NUM=\$(oc get node -lnode-role.kubernetes.io/master="" --no-headers | wc -l)
+HOSTEDCLUSTER_NAME=\$(cat /home/hostedcluster_name)
+echo "address=/api.\$HOSTEDCLUSTER_NAME.ostest.test.metalkube.org/192.168.111.\$((21+\$MASTER_NUM))" >> /etc/NetworkManager/dnsmasq.d/openshift-ostest.conf
```
Comment (eranco74) [inline ci-operator/step-registry/hypershift/agent/create/config-dns/hypershift-agent-create-config-dns-commands.sh:19]: I think it's better if you just take the IP of one of the cluster nodes:
```
oc get nodes master-0 -o jsonpath='{.status.addresses[0].address}'
```

File: ci-operator/step-registry/hypershift/agent/create/hostedcluster/hypershift-agent-create-hostedcluster-commands.sh (hunk 1 above)
Comment (LiangquanLi930) [reply ci-operator/step-registry/hypershift/agent/create/hostedcluster/hypershift-agent-create-hostedcluster-commands.sh:20]: yes

File: ci-operator/step-registry/hypershift/agent/create/config-dns/hypershift-agent-create-config-dns-commands.sh (hunk 2 above)
Comment (LiangquanLi930) [reply ci-operator/step-registry/hypershift/agent/create/config-dns/hypershift-agent-create-config-dns-commands.sh:19]: Sure, I think we can use this method, but can it use the master IP? I remember I made this attempt, worker ip is ready, but master ip is not to be ready

File: ci-operator/step-registry/hypershift/agent/create/hostedcluster/hypershift-agent-create-hostedcluster-commands.sh (hunk 1 above)
Comment (LiangquanLi930) [reply ci-operator/step-registry/hypershift/agent/create/hostedcluster/hypershift-agent-create-hostedcluster-commands.sh:20]: Ah, maybe we should make some adjustments, currently some hard code

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Comment by LiangquanLi930** [comment]
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/40143/rehearse-40143-periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance/1670376873911128064

File: ci-operator/step-registry/hypershift/agent/create/config-dns/hypershift-agent-create-config-dns-commands.sh (hunk 2 above)
Comment (eranco74) [reply ci-operator/step-registry/hypershift/agent/create/config-dns/hypershift-agent-create-config-dns-commands.sh]: Worker should be better

**Comment by eranco74** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-agent-ovn-conformance

**Review by eranco74 — COMMENTED** [review]
/lgtm

**Pull request merged**